### DISCUSSION TITLE
Class toggle on div when clicking an anchor with hash fragment

### DISCUSSION THREAD

**Discussion by thetutlage** [unpoly/unpoly#352] [General]
So, I have webpage that has anchor tags using hash fragment to link to div's within the same page. The anchor tags are not processed by unpoly, meaning they are not using `up-follow` or any other directive that triggers unpoly navigation on click.

However, everytime I click those links, the div on the same with the associated id gets an empty class attribute. For example, If I click `#home`, the div with `id=home` will get a class attribute.

Following is the recording of the behavior.

https://user-images.githubusercontent.com/1706381/154312400-55127ccd-c17e-4355-926f-e8c6529ae518.mov

- Live site link - https://beaded-erratic-sole.glitch.me
- Source code link - https://glitch.com/edit/#!/beaded-erratic-sole

**Comment by adam12** [comment]
Unpoly detects the `hashchange` and then adds and removes an `up-scrolling` class. Probably faster than you can see. This leaves the empty `class` attribute.

Visible if you set up a mutation breakpoint for attribute changes on that element.

Is the empty `class` attribute causing issues?

**Reply by thetutlage** [reply]
> Is the empty class attribute causing issues?

Yes, it is. Lemme share another screen recording. 

- In this one, when I click on the `about` link, you will see that the div gets hidden behind the sticky header.
- When I disable JavaScript (which inturn disables Unpoly), clicking the anchor keeps the div right below the sticky header. I am using `scroll-margin-top` property for this.

I do understand that Unpoly is not messing up with this CSS property. But somehow adding and removing that class is preventing the browser from that doing its thing.

I hope I was clear enough :)

https://user-images.githubusercontent.com/1706381/154785296-52a2f795-e19f-4839-b6ef-6cd18a7bb995.mov

**Comment by thetutlage** [comment]
Following up on this.

After digging into the source code, I found that the `revealHash` method is invoked every time I click on a jump to id link (even though Unpoly is not following the link).

Following is the source of `revealHash` method as of today.

```ts
    function revealHash(hash = location.hash, options) {
        let match = firstHashTarget(hash, options);
        if (match) {
            return up.reveal(match, { top: true });
        }
    }
```

The `up.reveal` call lead me to https://unpoly.com/up.reveal documentation, which lead me to `up-fixed` attribute.

Finally adding `up-fixed="top"` to my fixed header solved the issue.
